### DISCUSSION TITLE
Web Push API not making post to endpoint given on subscription creation

### DISCUSSION THREAD

**Discussion by vhawley** [mastodon/mastodon#22676] [Q&A]
### Steps to reproduce the problem

1. Create subscription by POSTing to `/api/v1/push/subscription`
2. Receive 200 response back with confirmed subscription id, endpoint, and subscribed alerts.
3. Trigger relevant event

Attempted on multiple instances (mastodon.social and mas.to) 

### Expected behaviour

Endpoint given when creating subscription should receive a POST request

### Actual behaviour

Endpoint does not receive any request

### Detailed description

I received this response from the API when hitting /api/v1/push/subscription.  However, when triggering something that should send a notification for the account specified on the token, nothing is received at the endpoint.  

I have confirmed this endpoint can receive POST requests (right now all it does is log the request received internally).  Checking my nginx logs, it doesnt seem to receive any kind of request request at any endpoint on this server when triggering an event.  

```
{
    "id": 908687,
    "endpoint": "https://api.victorhawleyjr.com/mastodon/notification/vtorlistener",
    "alerts": {
        "mention": true,
        "status": true,
        "reblog": true,
        "follow": true,
        "follow_request": true,
        "favourite": true,
        "poll": true,
        "update": true
    },
    "server_key": {redacted}
}
```

### Specifications

mastodon.social / mas.to
nginx 1.18.0 logging requests and reverse proxying an C# .NET API server.

**Comment by nightpool** [comment]
Are there any errors in the Sidekiq log? That would be the first place to start debugging this.

**Reply by vhawley** [reply]
Sidekiq would be for the instance operator, correct? I'm not using Ruby in my backend. I should have clarified that I was developing this against instances that are not mine.  I haven't tried spinning up my own Mastodon instance to debug this but maybe that's the next step diagnosing this.

**Reply by vhawley** [reply]
After trying with my own instance, I can see the logs on the instance side to see what's going on.  When I trigger an event, an error is thrown during the push job about invalid encoding on the p256v1 key.  so I must have done something wrong generating the keypair when creating the subscription.  Perhaps the key's encoding could be checked before the subscription is confirmed and return a 400 instead? 

```
Dec 23 15:44:58 mastodon402onubuntu2004-s-1vcpu-2gb-sfo3-01 bundle[702]: 2022-12-23T15:44:58.678Z pid=702 tid=2erq WARN: {"context":"Job raised exception","job":{"retry":5,"queue":"push","class":"Web::PushNotificationWorker","args":[1,2],"jid":"d8e9c9a5727bbdd7b58a>
Dec 23 15:44:58 mastodon402onubuntu2004-s-1vcpu-2gb-sfo3-01 bundle[702]: 2022-12-23T15:44:58.678Z pid=702 tid=2erq WARN: OpenSSL::PKey::EC::Point::Error: EC_POINT_bn2point: invalid encoding
Dec 23 15:44:58 mastodon402onubuntu2004-s-1vcpu-2gb-sfo3-01 bundle[702]: 2022-12-23T15:44:58.679Z pid=702 tid=2erq WARN: /home/mastodon/live/vendor/bundle/ruby/3.0.0/bundler/gems/webpush-f14a4d52e201/lib/webpush/encryption.rb:16:in `initialize'
```

**Reply by shaun0024** [reply]
Hey @vhawley, hope you're well. I came across this thread as I was coming across a similar issue. When I do a post to subscribe events to my endpoint, I get a 200 response but nothing seems to be sent either. I've also noticed that doing a get to the api to see details of my subscription results in a 404 not found response. Did you get yours working in the end? I am testing using the mas.to instance.
